DOC-5172: Added Corda5 topology diagrams for AWS and Azure (#1433)
corda/corda-docs-portal: content/en/platform/corda/5.0/deploying-operating/topology/_index.md

    weight: 3000
section_menu: corda5
---
# Infrastructure Topology

> The following topic shows topology diagrams for both Corda in AWS and Corda in Azure.

Reviewer: ```suggestion
This topic shows topology diagrams for both Corda in AWS and Corda in Azure.
```

Reviewer 2: Done.

> The size of the Kubernetes cluster required is dependent on the workload that the Corda cluster needs to handle. The Corda configuration section below gives some guidance on initial resource requests/limits to apply to the Corda workers which, when combined with the number of replicas for each worker type, can be used to estimate the total resources required in the Kubernetes cluster. The number of nodes in the cluster, and their distribution across availability zones, should take into account requirements for the availability of the solution.

Reviewer: Should the second sentence cross-ref this? https://corda5.preview.docs.r3.com/en/platform/corda/5.0/deploying-operating/deployment/deploying.html#resource-requests-and-limits

Reviewer 2: Yep - had added that after your review after talking with David.

Performance testing of your CorDapps under expected loads is required to determine the values that you will require in a production deployment.

> 

Reviewer: Should we have a short sentence introducing each diagram?

    * Azure resource provided by "Load balancing/Load Balancers", or "Network security groups".
    * Create your own, or use ones created at stage 5.
3.  Load Balancer:
    * Represents a load balancer that manages ingress traffic.
    * AWS resource provided by "EC2/Load Balancers".
    * Azure resource provided by "Load balancing/Load Balancers".
    * Create your own, or use one created at stage 5.
4.  Application Ingress Firewall:
    * Represents the network rules used to limit which networks are allowed to access the application network ingress.
    * AWS resource provided by "EC2/Security Groups".
    * Azure resource provided by "Network security groups".
    * Create your own, or use ones created at stage 5.
5.  Application Ingress:
    * Represents the Kubernetes Ingress Controller, or Kubernetes Service that exposes the cluster applications to the outside network.
    * Usually accompanied with a cloud-native load balancer with the ability to configure further.
    * Kubernetes resources provided by, but not limited to, "Ingress-Nginx", "Traefik Proxy", and the Kubernetes resource kind "Service" (type: LoadBalancer).
6.  Corda 5:

>     * TBC

Reviewer: Should probably be removed?

Reviewer 2: Done.

7.  Database Firewall:
    * Represents the network rules used to limit which networks are allowed to access the database.
    * AWS resource provided by "EC2/Security Groups".
    * Azure resource provided by "Network security groups".
8.  Database:

>     * Represents the datastore used by Corda 5.

Reviewer: ```suggestion
    * Represents the datastore used by Corda.
```

    * Supported database engine is "PostgreSQL" (version 14.4).
    * AWS resource provided by "RDS".
    * Azure resource provided by "Azure Database for PostgreSQL flexible servers".
9.  Egress Firewall (optional limitation):
    * Represents the network rules used to limit egress network access to external networks.
    * AWS resource provided by "EC2/Security Groups".
    * Azure resource provided by "Load balancing/Load Balancers", or "Network security groups".
10. Egress:
    * Represents the route external egress network traffic takes from the application network.

>     * In AWS it is routed (from Subnet: B) via a NAT gateway, and, if the destination is on the Internet, an Internet gateway.

Author: > * In AWS it is routed (from Subnet: B) via a NAT gateway, and, if the destination is on the Internet, an Internet gateway.

"Subnet: B" was updated to "Subnet: Private" in the diagrams after review from @JasonSinclair95
